cmd/dex/serve.go

```diff
@@ -197,7 +197,11 @@ func runServe(options serveOptions) error {
 				if client.Secret != "" {
 					return fmt.Errorf("invalid config: Secret and SecretEnv fields are exclusive for client %q", client.ID)
 				}
-				c.StaticClients[i].Secret = os.Getenv(client.SecretEnv)
+				secret := os.Getenv(client.SecretEnv)
+				if secret == "" {
+					return fmt.Errorf("invalid config: could not find SecretEnv %q", client.SecretEnv)
```

Comment:
Yeah, it should be breaking because if the env is undefined then the config is broken and not working anyway.

Reply:
1. Client secret can be empty if this is a public client.
2. If there is a single invalid client in the config, it will break all the working clients.

I insist on at least making this behavior opt-in.

Reply:
> Client secret can be empty if this is a public client.

That is unchanged. Only if the env used in secretEnv is empty or does not exist the  config would break. I consider this a hard configuration error.

For example the following config is invalid because envs cannot contain dashes:
```
- id: oauth2_proxy
  name: example
  redirectURIs:
  - https://example.de/oauth2/callback
  secretEnv: DEX_CLIENT_oauth2-proxy
```

also if secretEnv is set to a variable which wasn't provided, the config would fail. Before it just silently continued.

> If there is a single invalid client in the config, it will break all the working clients.

It would probably require more refactoring of the client loading logic, to skip invalid clients.

> I insist on at least making this behavior opt-in.

Should we allow end users to run with known broken configs? I am not sure what the security implications are when you run with a public client but intended to run with a secret. It probably breaks the security expectations and all clients being broken by this where broken anyway and didn't work correct.